lib/zenoss/events/event.rb

```diff
@@ -25,11 +26,25 @@ class Event < OpenStruct
       # Initialize this object from a Hash returned via the JSON api
       # @param[Zenoss] zenoss the current instance we are connecting with
       # @param[Hash] zhash a hash of values used to create this Event instance
+      # @return[Time] self.firstTime, self.lastTime
       def initialize(zenoss,zhash)
         @zenoss = zenoss
         super zhash
-        self.firstTime = DateTime.parse(self.firstTime) if self.firstTime
-        self.lastTime  = DateTime.parse(self.lastTime) if self.lastTime
+        #parse_time_format
+        self.firstTime &&= convert_time(self.firstTime)
+        self.lastTime &&= convert_time(self.lastTime)
+      end
+
+      # Converts a string or float to a Time object
+      # Zenoss version 4 emits the time format to be a string
+      # Zenoss version 6 emits the time to be a float
```

Comment:
I think grammatically we should have "as a" rather than "to be a" on these 2 lines.